### ISSUE TITLE
Support string filtering on top of semver tag filtering

### ISSUE THREAD

**Issue by Junkern** [xelalexv/dregsy#72]
Problem: I want to sync all node images with semver constraints `>=12.19.x || >=14.15.x` and only if the tag name contains `alpine` or `-buster`.
This is currently not possible with the [tag filtering ](https://github.com/xelalexv/dregsy#tag-filtering).

One could use the regex feature:
```
- 'regex: 12\.19\.[0-9]-(alpine|buster)'
- 'regex: 14\.15\.[0-9]-(alpine|buster)'
```
however for every new node minor version we would have to modify that list so that it stays up to date.
Is there any way to achieve the desired outcome with the current tag filtering?

**Comment by xelalexv** [comment]
An idea for improvement: If we introduce a tag filter `prune:`, which is a regex that gets applied last on the result of tag filtering, then your use case could be written like this:

```yaml
tags:
- semver: >=12.19.x || >=14.15.x
- prune: .+-(alpine|buster)
```

**Comment by Junkern** [comment]
Oh, this looks really neat and looks exactly what I need.
One suggestion: Could it be named `only` insteead of `prune`? Makes it more explicit what is being done with that filter. On the first sight it reads like all images which apply to `.+-(alpine|buster)` are being removed from the result set.
Do you need help in reviewing the https://github.com/xelalexv/dregsy/compare/issue72 branch?

**Comment by xelalexv** [comment]
I wasn't sure myself about `prune`, I agree that it may be a bit misleading. How about `keep`?

**Comment by Junkern** [comment]
`keep` also sounds like a really good option!

**Comment by xelalexv** [comment]
implemented with PR #77

**Comment by Junkern** [comment]
Thanks a lot @xelalexv !
Is there an ETA on a new release?

**Comment by xelalexv** [comment]
I'm still on vacation, but could probably create a new release next week.

**Comment by xelalexv** [comment]
released with `0.4.3` just now
